arch/arm/src/cxd56xx/cxd56_vectors.c

```diff
@@ -0,0 +1,107 @@
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include "chip.h"
+#include "arm_internal.h"
+#include "ram_vectors.h"
+#include "nvic.h"
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/* Chip-specific entrypoint */
+
+extern void __start(void);
+
+static void start(void)
+{
+  /* Zero lr to mark the end of backtrace */
+
+  asm volatile ("mov lr, #0\n\t"
+                "b  __start\n\t");
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/* Common exception entrypoint */
+
+extern void exception_common(void);
+extern void exception_direct(void);
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define IDLE_STACK      (_ebss + CONFIG_IDLETHREAD_STACKSIZE)
+
+/****************************************************************************
+ * Public data
+ ****************************************************************************/
+
+/* The cx56 use CXD56_IRQ_SMP_CALL to do SMP call.
+ * When sig dispatch do up_schedule_sigaction, need to make a new frame to
+ * run arm_sigdeliver. But the exception_direct cannot handle xcp.regs as we
+ * did not update the regs when c-function expection handler is called.
+ *
+ * We need to use exception_common to handle SMP call.
+ */
+
+const void * const cxd56_vectors[] locate_data(".vectors")
+  aligned_data(VECTAB_ALIGN) =
+{
+  /* Initial stack */
+
+  IDLE_STACK,
+
+  /* Reset exception handler */
+
+  start,
+
+  /* Vectors 2 - n point directly at the generic handler */
+
+  [2 ... NVIC_IRQ_PENDSV] = &exception_common,
+  [NVIC_IRQ_SYSTICK ... (CXD56_IRQ_SMP_CALL - 1)]
+                          = &exception_direct,
+  [CXD56_IRQ_SMP_CALL]    = &exception_common,
```

Comment:
can we reuse the common vector table?

Reply:
there is three extra chip have arm-m & smp support, the irq number will not easy to be a configure in Kconfig, so in update patch, make a custom vectors option to allow chip not use default vector.